htdocs/core/tpl/objectline_edit.tpl.php

```diff
@@ -303,6 +303,76 @@
 
 <script type="text/javascript">
 
+<?php
+if (! empty($usemargins) && $user->rights->margins->creer)
+{
+?>
+	/* Some js test when we click on button "Add" */
+	jQuery(document).ready(function() {
+		<?php
+		if (! empty($conf->global->DISPLAY_MARGIN_RATES)) { ?>
+		$("input[name='np_marginRate']:first").blur(function(e) {
+			return checkFreeLine(e, "np_marginRate");
+		});
+		<?php
+		}
+		if (! empty($conf->global->DISPLAY_MARK_RATES)) { ?>
+		$("input[name='np_markRate']:first").blur(function(e) {
+			return checkFreeLine(e, "np_markRate");
+		});
+		<?php
+		}
+		?>
+	});
+
+	/* TODO This does not work for number with thousand separator that is , */
+	function checkFreeLine(e, npRate)
+	{
+		var buying_price = $("input[name='buying_price']:first");
+		var remise = $("input[name='remise_percent']:first");
+
+		var rate = $("input[name='"+npRate+"']:first");
+		if (rate.val() == '')
+			return true;
+
+		if (! $.isNumeric(rate.val().replace(',','.')))
```

Comment:
Why don't you use price2numjs to get a valid value whatever is language locale ?

Reply:
@lvessiller-opendsi Still one question: Why don't you use price2numjs to get a valid numeric value whatever is language locale ?